<launch>
    <arg name="debug" default="false"/>
    <env if="$(arg debug)" name="ROSCONSOLE_CONFIG_FILE" value="$(find svcpp)/launch/debug.conf"/>

    <arg name="v" default="0"/>
    <env name="GLOG_v" value="$(arg v)"/>

    <arg name="tbb" default="0"/>
    <arg name="motion_comp" default="True"/>
    <arg name="log" default="5"/>
    <arg name="vis" default="0"/>
    <arg name="est" default="true"/>
    <arg name="odom_frame" default="odom"/>
    <arg name="world_frame" default="world"/>
    <arg name="body_frame" default="body"/>
    <arg name="min_path_dist" default="0.01"/>

    <node pkg="svcpp" type="sv_rofl_node_odom" name="rofl_odom" output="screen" ns="os_node">
        <remap from="~imu" to="imu"/>
        <remap from="~scan/image" to="image"/>
        <remap from="~scan/camera_info" to="camera_info"/>

        <rosparam command="load" file="$(find svcpp)/config/rofl.yaml" />
        <param name="log" type="int" value="$(arg log)"/>
        <param name="vis" type="int" value="$(arg vis)"/>
        <param name="est" type="bool" value="$(arg est)"/>

        <param name="odom/tbb" type="int" value="$(arg tbb)"/>
        <param name="traj/motion_comp" type="bool" value="$(arg motion_comp)"/>
        <param name="odom_frame" type="string" value="$(arg odom_frame)"/>
        <param name="world_frame" type="string" value="$(arg world_frame)"/>
        <param name="body_frame" type="string" value="$(arg body_frame)"/>
        <param name="min_path_dist" type="double" value="$(arg min_path_dist)"/>
    </node>
</launch>
